File: app/src/main/java/com/jobinlawrance/pics/di/fragment/FragmentKey.java
package com.jobinlawrance.pics.di.fragment;


import android.support.v4.app.Fragment;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

import dagger.MapKey;

/**
 * Created by jobinlawrance on 22/9/17.
 */
@MapKey
@Target(ElementType.METHOD)
public @interface FragmentKey {
    Class<? extends Fragment> value();
}

File: app/src/main/java/com/jobinlawrance/pics/di/fragment/FragmentKey.kt
package com.jobinlawrance.pics.di.fragment


import android.support.v4.app.Fragment

import dagger.MapKey
import kotlin.reflect.KClass

/**
 * Created by jobinlawrance on 22/9/17.
 */
@MapKey
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class FragmentKey(val value: KClass<out Fragment>)
